lightkurve/targetpixelfile.py

```diff
@@ -861,33 +923,73 @@ def from_fits_images(images, position=None, size=(10, 10), extension=None,
         tpf : KeplerTargetPixelFile
             A new Target Pixel File assembled from the images.
         """
+        from lightkurve.targetpixelfile import _get_pos_corrs
         if extension is None:
             if isinstance(images[0], str) and images[0].endswith("ffic.fits"):
                 extension = 1  # TESS FFIs have the image data in extension #1
             else:
                 extension = 0  # Default is to use the primary HDU
 
+
         factory = KeplerTargetPixelFileFactory(n_cadences=len(images),
                                                n_rows=size[0],
                                                n_cols=size[1],
                                                target_id=target_id)
+        # Find middle image to use as a reference
+        mid_img = images[int(len(images)/2)-1]
+
         for idx, img in tqdm(enumerate(images), total=len(images)):
+            #Check if an image is a single FITS image, or a FITS file, and return the appropriate PrimaryHDU or ImageHDU.
             if isinstance(img, fits.ImageHDU):
                 hdu = img
             elif isinstance(img, fits.HDUList):
                 hdu = img[extension]
             else:
                 hdu = fits.open(img)[extension]
+
             if idx == 0:  # Get default keyword values from the first image
                 factory.keywords = hdu.header
             if position is None:
                 cutout = hdu
             else:
                 cutout = Cutout2D(hdu.data, position, wcs=WCS(hdu.header),
                                   size=size, mode='partial')
-            factory.add_cadence(frameno=idx, flux=cutout.data, header=hdu.header)
-        return factory.get_tpf(**kwargs)
-
+            pos_corr1, pos_corr2, col, row, ractr, decctr = \
+                _get_pos_corrs(mid_img, hdu, position, extension)
+            # Add calculated pos corr shifts to the keywords dictionary
+            hdu.header['POSCORR1'] = pos_corr1
+            hdu.header['POSCORR2'] = pos_corr2
+
+            factory.add_cadence(frameno=idx, wcs=WCS(hdu.header), flux=cutout.data, header=hdu.header)
+
+        tpfsize = size[0] # Use column value assuming a square TPF
+        if (tpfsize % 2 == 0): # Make sure tpfsize is odd to center the star
+             tpfsize += 1
+
+        # Override the defaults where necessary, and pass into the header generating functions
+        ext_info = {}
+        ext_info['TFORM4'] = '{}J'.format(size[0]*size[1])
+        ext_info['TDIM4'] = '({},{})'.format(size[0],size[1])
+
+        cd11 = factory.keywords['CD1_1']
+        cd12 = factory.keywords['CD1_2']
+        cd21 = factory.keywords['CD2_1']
+        cd22 = factory.keywords['CD2_2']
```

Comment:
It looks to me here like you may be takign the cd* values from the last fits image and not the middle one? I would find a way to just read in the middle image's info- maybe another stand-alone function. Because you need to extract both the CD* keywords, as well as figure out ractr, dectr. This should only require one call.